http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/api/CorsSupport.scala

```diff
@@ -23,20 +27,49 @@ class CorsSupport(contentSecurityPolicy: ContentSecurityPolicyConfig) {
     )
   }
 
+  /** Whether to allow any Origin header value or to restrict to the allowlist.
+   * Enable strict mode by either removing the wildcard (*) from
+   * refererConfig.validHosts or setting refererConfig.originStrict to true.  */
+  private val isOriginStrict: Boolean = !refererConfig.validHosts.contains("*") || refererConfig.originStrict
+
+  private val handleOrigin: Directive0 =
+    if (!refererConfig.enabled || !isOriginStrict) pass
+    else
+      optionalHeaderValueByType(Origin) flatMap {
+        case Some(origin) =>
+          logger.info(s"CorsSupport | Request origin detected: ${origin}. Checking validity")
+          checkSameOrigin(getValidOriginRange)
+        case None =>
+          logger.info(s"CorsSupport | Request origin not provided. Permitting")
+          pass
+      }
+
+  private def getValidOriginRange: HttpOriginRange.Default = {
+    def validOrigins: Set[HttpOrigin] = refererConfig.validHosts
+      .filter(_ != "*")
+      .map(uri => if (uri.last == '/') uri.slice(0, uri.length - 1) else uri)
+      .flatMap { uriString =>
+        Set(HttpOrigin(s"http://${uriString}"), HttpOrigin(s"https://${uriString}"))
```

Comment:
Curious if adding both schemes is necessary, or if you could just do `HttpOrigin(uriString)` and akka would match it.

Reply:
Needed! the `uriString` here omits the scheme, `HttpOrigin` requires it, and I wasn't sure which of http/https we'd be expecting (and doing one doesn't automatically allow the other)